docs/library/machine.Counter.rst

```diff
@@ -0,0 +1,77 @@
+.. currentmodule:: machine
+.. _machine.Counter:
+
+class Counter-- Pulse Counter
+=============================
+
+This class provides a Pulse Counter service.
+
+Here is described a basics, commons for hardware-counters-based counters of MicroPython ports:
+
+  * :ref:`ESP32.Counter <pcnt.Counter>`
+  * mimxrt.Counter - under constructions
+  * STM32.Counter - TODO
+
+Minimal example usage::
+
+    from machine import Pin, Counter
+
+    counter = Counter(0, Pin(0, mode=Pin.INPUT))  # create Counter object and start to count input pulses
+    counter.init(filter_ns=0)                     # switch filtering off
+    value = counter.value()                       # get current Counter value
+    value = counter.value(0)                      # get current Counter value, set counter to 0
+    counter.deinit()                              # turn off the Counter
+
+    print(counter)                                # show the Counter object properties
+
+Constructor
+-----------
+
+.. class:: Counter(id, src=machine.Pin, \*, keyword_arguments)
```

Comment:
Using the `=` implies that there is a reasonable default value and the parameter does not have to be set. But that is not the case. Semantically it is a mandatory parameter for the constructor. So I prefer:
`Counter(id, src, *, direction, filter_ns)`. For the init() call, it is optional, since it had been set already by the constructor.

Reply:
The very best way to write documents is to go and look at the existing ones and copy the style.

Also, I think `src` *could* be optional in specific ports if the pins are tied to the hardware units.